pkg/cfn/builder/nodegroup.go

```diff
@@ -370,9 +370,9 @@ func metricsCollectionResource(asgMetricsCollection []api.MetricsCollection) []m
 	for _, m := range asgMetricsCollection {
 		newCollection := make(map[string]interface{})
 
-		var metrics []string
-		metrics = append(metrics, m.Metrics...)
-		newCollection["Metrics"] = metrics
+		if len(m.Metrics) > 0 {
+			newCollection["Metrics"] = m.Metrics
```

Comment:
The bug is that we have `newCollection["Metrics"] = nil`, right? `[]` seems to be a valid value for CFN.

I think comparing to `nil` should be enough?

Reply:
Thanks for your input. I am not sure about the effect of passing [] in CF, will it fail for validation, or not ? So to be safe, I just proceed to with _len()_ (which internally check for nil slice as well).

If you have a chance to confirm the above, I can make a chance. Result will be the same from end user 's point of view.

Reply:
I guess the question is what does the user expect in the case where they pass `[]` explicitly, no metrics or all metrics?

Reply:
I can't test it out right now as I have no credit in my aws account. My intepretation on aws docs for this case is _all metrics_

Reply:
indeed, seems to be the case that `null` and `[]` are treated the same by cfn.